src/pytorch_lightning/strategies/colossalai.py

```diff
@@ -0,0 +1,294 @@
+
+        Returns: Model parallel context.
+        """
+
+        class ModelShardedContext(ColoInitContext):
+            def _post_init_method(self, module: torch.nn.Module, *args, **kwargs):
+                if getattr(module, "_colossalai_module", False) is True:
+                    return
+                super()._post_init_method(module, *args, **kwargs)
+                module._colossalai_module = True
+
+        return ModelShardedContext()
+
+    def setup_precision_plugin(self) -> None:
+        super().setup_precision_plugin()
+        is_training = self.lightning_module.trainer and self.lightning_module.trainer.training
+        if is_training:
+            if len(self.optimizers) > 1:
+                raise MisconfigurationException("ColossalAIStrategy only supports single Optimizer now.")
+            optimizer = self.optimizers[0]
+            assert isinstance(optimizer, (CPUAdam, HybridAdam)), (
+                "ColossalAIStrategy only supports colossalai.nn.optimizer.CPUAdam "
+                "and colossalai.nn.optimizer.HybridAdam."
+            )
+        pl_module = self.model
+        if not hasattr(pl_module, "_colossalai_zero"):
+            if self.use_chunk:
+                chunk_size = self.chunk_size or ChunkManager.search_chunk_size(
+                    self.model, **self.chunk_size_search_kwargs
+                )
+            else:
+                chunk_size = None
+            chunk_manager = ChunkManager(
+                chunk_size,
+                self.process_group,
+                self.enable_distributed_storage,
+                GeminiManager.get_default_device(self.placement_policy),
+            )
+            gemini_manager = GeminiManager(self.placement_policy, chunk_manager)
+            assert isinstance(self.model, (pl.LightningModule, _LightningPrecisionModuleWrapperBase))
+            model = _LightningModuleWrapperBase(self.model)
+            self.model = ZeroDDP(model, gemini_manager, self.force_outputs_fp32)
+            pl_module._colossalai_zero = [self.model]
+        else:
+            self.model = pl_module._colossalai_zero[0]
+        if is_training:
+            self.optimizers = [
+                ZeroOptimizer(optimizer, self.model, gpu_margin_mem_ratio=self.gpu_margin_mem_ratio, **self.amp_kwargs)
+            ]
+
+    def setup(self, trainer: "pl.Trainer") -> None:
+        if not isinstance(self.accelerator, CUDAAccelerator):
+            raise MisconfigurationException(
+                f"Colossalai is only supported on GPU but `{self.accelerator.__class__.__name__}` is used."
+            )
+        self.accelerator.setup(trainer)
+
+        assert self.lightning_module is not None
+        self.lightning_module._device = self.root_device
+
+        if trainer.accumulate_grad_batches > 1:
+            raise MisconfigurationException(
+                "Colossalai does not support gradient accumulation now. " "please set `accumulate_grad_batches` to 1."
```

Comment:
There is a problem with the string here:
```suggestion
                "Colossalai does not support gradient accumulation now. Please set `accumulate_grad_batches` to 1."
```

Reply:
Fixed now.